lib/pbench/test/unit/server/test_datasets_compare.py

```diff
@@ -92,16 +96,27 @@ def mock_compare_csv_to_json(*args, **kwargs):
             nonlocal extract_data_called
             extract_data_called = True
 
-        def mock_get_value(dataset, key) -> str:
-            return "hammerDB"
+        class MockTarball:
+            tarball_path = Path("/dataset/tarball.tar.xz")
+            name = "tarball"
 
-        monkeypatch.setattr(CacheManager, "find_dataset", self.mock_find_dataset)
+            def extract(_tarball_path: Path, _path: str) -> str:
+                return "IncorrectData"
+
+        def mock_find_dataset_with_incorrect_data(self, dataset) -> MockTarball:
+            # Validate the resource_id
+            Dataset.query(resource_id=dataset)
+            return MockTarball
```

Comment:
Ah, yes: as Webb pointed out elsewhere, `return MockTarball()` would be correct.

Reply:
Yes...but it'll probably force him to correct something else....  😁